### PR TITLE
[Docs] Add initial CHANGELOG.md with changelogs from previous release

### PR DESCRIPTION
**Pull request by AdarshRawat1**
# What does change log contain:
- A summary of major changes
- Details on new features
- Bug fixes
- Other notable improvements and modifications 

## Impact:
- This changelog will be available on the documentation site. 
- Users and contributors can easily track changes and updates over time through the documentation site.
## Screenshot
image

## Reference 
Based on this discussion - https://github.com/p4lang/p4c/pull/4692#issuecomment-2140166182

## Other changes 
bumped up `p4c_layout.xml` v1.8.12 -->v1.11.0

### CODE REVIEW

**Comment by fruffy** [comment]
Is this ready for review?

**Comment by AdarshRawat1** [comment]
> Is this ready for review?

I'm formatting the previous release notes as per the current release format.[i.e. Under categories - Breaking changes , p4 specific implementation etc]

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3240 @@
+<!-- \page changelog Changelog -->
```
Comment (fruffy) [inline docs/CHANGELOG.md:1]: I would remove all the commits before the earliest known release. They are just noise. We can add a note that only the commit history is available for those.

**Review by fruffy — CHANGES_REQUESTED** [review]

File: docs/CHANGELOG.md (hunk 1 above)
Comment (AdarshRawat1) [reply docs/CHANGELOG.md]: I added notes for semantics and previous commits, though they do need some polishing.

Comment (AdarshRawat1) [reply docs/CHANGELOG.md]: > I would remove all the commits before the earliest known release. They are just noise. We can add a note that only the commit history is available for those.

So, will we remove this latest one as well, when we get the latest release or will we append the newer release to the list.

**Comment by AdarshRawat1** [comment]
Screenshot of current changelog configuration [With awesome CSS] -[ view live here ](https://adarshrawat1.github.io/P4c-Documentation-Hosting/)

image

File: docs/CHANGELOG.md (hunk 1 above)
Comment (fruffy) [reply docs/CHANGELOG.md]: I was looking at https://github.com/AdarshRawat1/p4c/blob/Docs-Changelogs/CHANGELOG.md

The version that included previous releases was fine, but that list only went until [v1.2.2](https://github.com/p4lang/p4c/releases/tag/v1.2.2). We can keep the changelog until that point, then delegate to the git commit history before that.

**Comment by fruffy** [comment]
> Screenshot of current changelog configuration [With awesome CSS] -[ view live here ](https://adarshrawat1.github.io/P4c-Documentation-Hosting/)
> 
> image

Looks great!

**Comment by Dscano** [comment]
Looks good to me

File: docs/CHANGELOG.md (hunk 1 above)
Comment (AdarshRawat1) [reply docs/CHANGELOG.md]: Oh Thanks !! , I understand it now.

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,3272 @@
+\page Changelog Changelog
+## Semantic Versioning
+We follow a monthly release cadence. Our versioning scheme is as follows:
+- **Major.Minor.Patch** versions align with the P4 specification.
+- **Revision** (the last number) is incremented with each monthly release.
+
+> [!TIP]
```
Comment (fruffy) [inline CHANGELOG.md:7]: Call this Note instead of tip.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,3272 @@
...
+- Add correct and error test cases for structure-values expressions [[view](https://github.com/p4lang/p4c/pull/3100)]
+- Treat match_kind as a regular type [[view](https://github.com/p4lang/p4c/pull/3099)]
+- dpdk: Eliminate unused metadata fields [[view](https://github.com/p4lang/p4c/pull/3096)]
+- Preserve some types in constant folding. [[view](https://github.com/p4lang/p4c/pull/3094)]
+- Emit semicolon after method call [[view](https://github.com/p4lang/p4c/pull/3093)]
+- Dismantle Mux expression in dpdk [[view](https://github.com/p4lang/p4c/pull/3087)]
+- Moved FindRecirculated from .cpp to .h [[view](https://github.com/p4lang/p4c/pull/3088)]
+
+
+## Release 1.2.2.1 [[view](https://github.com/p4lang/p4c/pull/3085)]
```
Comment (fruffy) [inline CHANGELOG.md:1078]: Start to defer to the commit history here.

**Review by fruffy — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,1075 @@
+\page Changelog Changelog
```
Comment (fruffy) [inline CHANGELOG.md:1]: I missed this. Please remove then we should be ready to go.

Comment (fruffy) [inline CHANGELOG.md:1]:
```suggestion
```

Comment (AdarshRawat1) [reply CHANGELOG.md]: Done !

**Review by fruffy — APPROVED** [review]
Let's get this in, we can always improve minor problems later.

**Pull request merged**